ecl/eclagent/eclagent.cpp

```diff
@@ -564,7 +564,11 @@ EclAgent::EclAgent(IConstWorkUnit *wu, const char *_wuid, bool _checkVersion, bo
         updateSuppliedXmlParams(w);
     }
     IPropertyTree *costs = queryCostsConfiguration();
-    workflowMachineCost = costs ? costs->getPropReal("@workflowMachine", 0.0): 0.0;
+    if (costs)
+    {
+        workflowMachineCost = costs->getPropReal("@workflowMachine", 0.0);
```

Comment:
I don't think you can differentiate between these two.  different agents may have different costs, but within a single process there is only going to be one cost.